### ISSUE TITLE
"font-family-no-duplicate-names" - broken

### ISSUE THREAD

**Issue by dvlden** [stylelint/stylelint#3368]
**Describe the bug? (A clear and concise description of what the bug is)**

Rule: `font-family-no-duplicate-names` is not working as described when set to `true` for scss syntax.

**Which rule, if any, is this issue related to?**

`font-family-no-duplicate-names`

**What CSS is needed to reproduce this issue?**

> This example is working... 

```css
.foo {
  font-family: "Helvetica", Helvetica, sans-serif;
}
```

> This example is not working...

```scss
.foo {
  font: {
     family: "Helvetica", Helvetica, sans-serif;
  }
}
```

**What stylelint configuration is needed to reproduce this issue?**

```json
{
  "rules": {
    "font-family-no-duplicate-names": true
  }
}
```

**Which version of stylelint are you using?**

`9.2.1`

**How are you running stylelint: CLI, PostCSS plugin, Node API?**

CLI with `stylelint --sytnax=scss \"src/**/**.scss\"`

**Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?**

Yes, it's related to Scss.

**What did you expect to happen?**

To get an error, such as: `✖  Unexpected duplicate name Helvetica`

**What actually happened (e.g. what warnings or errors you are getting)?**

Nothing, it passes the linting.

**Comment by hudochenkov** [comment]
Thanks for the report and for using the template.

Stylelint doesn't support Sass nested properties, that's why this rule doesn't recognize `family` as `font-family`.

**Comment by dvlden** [comment]
Gotcha! Is there any extension/plugin for this that you might be aware of?

**Comment by hudochenkov** [comment]
I'm afraid there isn't anything that will enable nested properties to work with stylelint core rules.

**Comment by dvlden** [comment]
Oh well, that is unfortunate. @hudochenkov 
But still, it already helps with so many things, one thing more/less won't do any harm. Decent contribution guide should be enough for those minor things that aren't supported.

Great work on the linter tho! So many rules, my head exploded by reading all of them today and writing the rules for my open-source project.

**Comment by jeddy3** [comment]
@dvlden There is also the option of contributing a `font-family-no-duplicate-names` rule to the [stylelint-scss](https://github.com/kristerkari/stylelint-scss) plugin pack. You could base it on the built-in rule, but attempt to add support for the nested properties non-standard construct. You'd then use that plugin rule, rather than the built-in one, in your configuration object.  This is [already possible](https://github.com/kristerkari/stylelint-scss#-rule) with rules like `at-rule-no-unknown`.
